Server/ReasnAPI/ReasnAPI/Services/StatusService.cs

```diff
@@ -1,16 +1,97 @@
 ﻿using ReasnAPI.Models.Database;
+using ReasnAPI.Models.DTOs;
+using System.Linq.Expressions;
 
-namespace ReasnAPI.Services {
-    public class StatusService (ReasnContext context) {
-        private readonly ReasnContext _context = context;
-
-        /* TODO: Create following functions for this class
-         * create
-         * update
-         * delete
-         * get by ID
-         * get list by filter
-         * get all
-         */
+namespace ReasnAPI.Services;
+public class StatusService(ReasnContext context)
+{
+    public StatusDto CreateStatus(StatusDto statusDto)
+    {
+        var status = context.Statuses.FirstOrDefault(r => r.Name == statusDto.Name);
+        if (status != null)
+        {
+            return null;
+        }
+        var newStatus = new Status
+        {
+            Name = statusDto.Name,
+            ObjectTypeId = statusDto.ObjectTypeId
+        };
+
+        context.Statuses.Add(newStatus);
+        context.SaveChanges();
+
+        return statusDto;
+    }
+
+    public StatusDto UpdateStatus(int statusId, StatusDto statusDto)
+    {
+        var status = context.Statuses.FirstOrDefault(r => r.Id == statusId);
+
+        if (status == null)
+        {
+            return null;
+        }
+
+        status.Name = statusDto.Name;
+        status.ObjectTypeId = statusDto.ObjectTypeId;
+
+        context.Statuses.Update(status);
+        context.SaveChanges();
+
+        return statusDto;
     }
-}
+
+    public bool DeleteStatus(int statusId)
+    {
+        var status = context.Statuses.FirstOrDefault(r => r.Id == statusId);
+
+        var statusCheck = context.Participants.FirstOrDefault(r => r.StatusId == statusId);
+        var statusCheck2 = context.Events.FirstOrDefault(r => r.StatusId == statusId);
```

Comment:
Change variable names to something more meaningful like statusAssignedToParticipants, statusAssignedToEvents etc.

Reply:
Changed